glyphs-reader/src/font.rs

```diff
@@ -1702,31 +1658,136 @@ impl TryFrom<RawFont> for Font {
+    let unicode_re =
+        Regex::new(r"(?m)^(?P<prefix>\s*unicode\s*=\s*)[(]?(?P<value>[0-9a-zA-Z,]+)[)]?;\s*$")
+            .unwrap();
+    // Some special glyph names can be parsed as floats so we enquote them
+    let glyphname_re =
+        Regex::new(r"(?mi)^(?P<prefix>\s*glyphname\s*=\s*)(?P<value>(?:infinity|inf|nan));\s*$")
+            .unwrap();
+    let s = unicode_re.replace_all(&s, r#"$prefix"$value";"#);
+    let s = glyphname_re.replace_all(&s, r#"$prefix"$value";"#);
+    s.into_owned()
+}
+
 impl Font {
     pub fn load(glyphs_file: &path::Path) -> Result<Font, Error> {
-        debug!("Read {glyphs_file:?}");
-        let raw_content = fs::read_to_string(glyphs_file).map_err(Error::IoError)?;
+        if glyphs_file.is_dir() {
+            return Font::load_package(glyphs_file);
+        }
 
-        // Glyphs has a wide variety of unicode definitions, not all of them parser friendly
-        // Make unicode always a string, without any wrapping () so we can parse as csv, radix based on format version
-        let re =
-            Regex::new(r"(?m)^(?P<prefix>\s*unicode\s*=\s*)[(]?(?P<value>[0-9a-zA-Z,]+)[)]?;\s*$")
-                .unwrap();
-        let raw_content = re.replace_all(&raw_content, r#"$prefix"$value";"#);
+        debug!("Read {glyphs_file:?}");
+        let raw_content =
+            preprocess_unparsed_plist(fs::read_to_string(glyphs_file).map_err(Error::IoError)?);
 
-        let mut raw_content = Plist::parse(&raw_content)
+        let raw_content = Plist::parse(&raw_content)
             .map_err(|e| Error::ParseError(glyphs_file.to_path_buf(), format!("{e:#?}")))?;
 
-        // Fix any issues with the raw plist
-        let Plist::Dictionary(ref mut root_dict) = raw_content else {
+        let raw_font = RawFont::from_plist(raw_content);
+        raw_font.try_into()
+    }
+
+    pub fn load_package(glyphs_package: &path::Path) -> Result<Font, Error> {
+        if !glyphs_package.is_dir() {
+            return Err(Error::NotAGlyphsPackage(glyphs_package.to_path_buf()));
+        }
+        debug!("Read {glyphs_package:?}");
+
+        let fontinfo_file = glyphs_package.join("fontinfo.plist");
+        let fontinfo_data = fs::read_to_string(&fontinfo_file).map_err(Error::IoError)?;
+        let mut font_plist = Plist::parse(&fontinfo_data)
+            .map_err(|e| Error::ParseError(fontinfo_file.to_path_buf(), format!("{e:#?}")))?;
+
+        let Plist::Dictionary(ref mut root_dict) = font_plist else {
             return Err(Error::ParseError(
-                glyphs_file.to_path_buf(),
+                fontinfo_file.to_path_buf(),
                 "Root must be a dict".to_string(),
             ));
         };
-        fix_glyphs_named_infinity(glyphs_file, root_dict)?;
 
-        let raw_font = RawFont::from_plist(raw_content);
+        let mut glyphs: HashMap<String, Plist> = HashMap::new();
```

Comment:
if we know this is always a plist dictionary we can just use the dictionary type directly? (I've added a `Dictionary` type in #450)

Reply:
this particular one has to be a Plist type, not the inner BTreeMap, because I'm inserting these in the "glyphs" plist array later on

Reply:
okay then another good thing to know is that the general 'Plist' type has a .get() method, which returns `None` if you call it on a non-dict.

Reply:
I ended up using your Dictionary alias, but then when pushing it to the glyphs array I need to wrap it again in a Plist::Dictionary enum variant